ios/browser/api/qr_code/qr_code_generator.mm

```diff
@@ -0,0 +1,107 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/qr_code/qr_code_generator.h"
+#include "base/strings/sys_string_conversions.h"
+#include "brave/ios/browser/qr_code_generator/qrcode_generator_service.h"
+#include "brave/ios/browser/qr_code_generator/qrcode_models.h"
+
+#include "ui/gfx/image/image.h"
+#include "ui/gfx/image/image_skia.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+// MARK: - Implementation
+
+// MARK: - BraveQRCodeGeneratorOptions
+@interface BraveQRCodeGeneratorOptions () {
+  std::unique_ptr<qrcode_generator::GenerateQRCodeRequest> request_;
+}
+@end
+
+@implementation BraveQRCodeGeneratorOptions
+- (instancetype)initWithData:(NSString*)data
+                shouldRender:(bool)shouldRender
+          renderLogoInCenter:(bool)renderLogoInCenter
+           renderModuleStyle:(BraveQRCodeGeneratorModuleStyle)renderModuleStyle
+          renderLocatorStyle:
+              (BraveQRCodeGeneratorLocatorStyle)renderLocatorStyle {
+  if ((self = [super init])) {
+    request_ = std::make_unique<qrcode_generator::GenerateQRCodeRequest>(
+        base::SysNSStringToUTF8(data), shouldRender, renderLogoInCenter,
+        static_cast<qrcode_generator::ModuleStyle>(renderModuleStyle),
+        static_cast<qrcode_generator::LocatorStyle>(renderLocatorStyle));
+  }
+  return self;
+}
+
+- (void)dealloc {
+  request_.reset();
+}
+
+- (qrcode_generator::GenerateQRCodeRequest*)nativeRequest {
+  return request_.get();
+}
+@end
+
+// MARK: - BraveQRCodeGeneratorResult
+@interface BraveQRCodeGeneratorResult () {
+  std::unique_ptr<qrcode_generator::GenerateQRCodeResponse> response_;
+}
+@end
+
+@implementation BraveQRCodeGeneratorResult
+- (instancetype)initWithResponse:
+    (std::unique_ptr<qrcode_generator::GenerateQRCodeResponse>)response {
+  if ((self = [super init])) {
+    response_ = std::move(response);
+  }
+  return self;
+}
+
+- (void)dealloc {
+  response_.reset();
+}
+
+- (BraveQRCodeGeneratorError)getErrorCode {
+  return static_cast<BraveQRCodeGeneratorError>(response_->error_code);
+}
+
+- (UIImage*)getImage {
+  if (!_image) {
+    gfx::Image image = gfx::Image(gfx::ImageSkia::CreateFromBitmap(
+        response_->bitmap, [[UIScreen mainScreen] scale]));
+    _image = image.IsEmpty() ? [[UIImage alloc] init] : image.ToUIImage();
+  }
+  return _image;
+}
+
+- (NSData*)getData {
+  if (!_data) {
+    _data = [NSData dataWithBytes:response_->data.data()
+                           length:response_->data.size()];
+  }
+  return _data;
+}
+
+- (CGSize)getDataSize {
```

Comment:
These getters all have `get` prefixes but the property definitions don't, so they won't be called? Probably need to remove all of the `get` prefixes to match

Reply:
Fixed. I was trying to optimize the call-stack there because calling it many times will be slow and I wanted to use as little memory as possible checking against if the `ivar` was set. But I realize this is dumb, and I can just set everything in the constructor, so I changed that just now and pushed :)